plugins/workspace-backpack/src/index.js

```diff
@@ -438,43 +439,48 @@ export class Backpack {
    *     cleaned of all unnecessary attributes.
    */
   addItem(item) {
-    if (this.contents_.indexOf(item) !== -1) {
-      return;
-    }
+    this.addItems([item]);
+  }
 
-    this.contents_.unshift(item);
-    while (this.contents_.length > this.maxItems_) {
-      this.contents_.pop();
-    }
-    // TODO: Fire UI event for Backpack content change.
+  /**
+   * Adds multiple items to the backpack.
+   * @param {!Array<string>} items The backpack contents to add.
+   */
+  addItems(items) {
+    const filteredItems = items.filter((item) => {
+      return this.contents_.indexOf(item) === -1;
```

Comment:
I've refactored, but I think the `contains` method is an interesting idea. 
Since this PR is supposed to be about Events, I don't want to update it here, but I can address in the context menu PR.
https://github.com/google/blockly-samples/pull/728